### ISSUE TITLE
Enable Community Stakes page

### ISSUE THREAD

**Issue by sachben91** [hicommonwealth/commonwealth#5878]
## User Story

After I have set up my community on Common, I should be directed to the third and optional step of the create community flow - Enabling Community Stakes. Here I should be provided with information about Community Stakes and prompted to enable it or opt out. If I click "yes" to enable stakes, the page should transition to information about the transactions that need to happen to enable Community Stakes. If I click "No" I should be directed to the "Community is Live" page. 

The figjam here shows the enabling stakes flow. As mentioned in the bucket ticket, the community should be created on Common after the figjam so that even if the transactions fail, the user can still access their community. 

![Image](https://github.com/hicommonwealth/commonwealth/assets/38763242/9a51b694-f4fe-4fda-b30c-fa8f8716b2e5)

## Product specs 

- [ ] namespace of the community on Common should not be case sensitive, special characters are allowed, space is not allowed. It should be autogenerated by default to be the community name that the user entered in basic information page. The user should be able to edit it if they want to. The namespace must be unique
- [ ] The symbol should be autogenerated using the first four letters of the Community name entered by the user. The symbol should be displayed in upper case. User should be able to edit it. The rules for symbols should be the same as it is right now 
- [ ] When the user clicks on "learn more.." it should redirect to a blog post (tbd) that explains stakes 
- [ ] When the user clicks "Yes", the page should transition to "sign transactions to launch stakes" 
- [ ] User should sign the transaction for the namespace first. Only after completing the signing of the namespace should they be able to sign the transaction to launch Community stakes 
- [ ] Clicking on sign initiates the wallet transaction
- [ ] It should be communicated to the user that they should not navigate away from the page as the transactions happen
- [ ] If the transaction fails, user should be shown the respective failed transaction error state messaging 
- [ ] Once transaction is complete, the user should be taken to the Community is Live page 

## Engineering specs
Refer to this Model for detailed engineering flow, noting relevant actions after user actions: https://miro.com/app/board/uXjVNZ8oB3U=/?moveToWidget=3458764572378160162&cot=14
1. Before sending namespace reservation transaction, check for namespace availability using contract helper methods and prompt user to change name until name is available(ie invalid text field)
2. Upon successful transaction for namespace reservation the route defined in this ticket should be called: https://github.com/hicommonwealth/commonwealth/issues/5882 
    - store and include the following data from the transaction for request: namespace name, network, transaction hash
3. Upon a successful community stakes configuration the route defined in this ticket should be called: https://github.com/hicommonwealth/commonwealth/issues/5884
    - store and include the following data from transaction for request: community stake id(from sdk), transaction hash

## Designs 

[Figma](https://www.figma.com/file/WLN4nRpVGQJ9AjuxiJwuDS/Create-Community?type=design&node-id=1027%3A7352&mode=dev)

**Comment by masvelio** [comment]
@sachben91 Questions:
1. > Namespace should be autogenerated by default to be the community name that the user entered in basic information page.

We will re-use a util that we use for generating community ID in previous step. ID is used for the community URL field. So basically, if community name is `Warriors` then ID is `warriors`. If community name is `Super Dao` then ID is `super-dao` etc.
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/a186ed00-1357-471c-aa59-141b2dd3adce)

2. > The rules for symbols should be the same as it is right now

The only validation we have right now is max 9 characters. This means we can default to 4 first letters but then user can modify it and type any other string with any alphanumeric character, special character and white spaces. Is this ok?

3. We have 4 different states in `Sign transactions to launch stakes` [shown here in order](https://www.figma.com/file/WLN4nRpVGQJ9AjuxiJwuDS/Create-Community?type=design&node-id=1019%3A2958&mode=dev). Does cancel button look & work the same way at every state? What happens on clicking cancel button? Maybe when the transaction is being signed we can disabled it, as it is tempting to click the button while waiting for something. Also, do we want to display any "warning modal" on clicking cancel eg when one of transaction has already signed or is being signed, similarly what we do for canceling second step 
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/c471294d-2129-4f84-ba8d-f37debdc1952)

4. We don't have currently this type of loader in the app. This is `CircleNotch` phosphor icon. Should it just spin? Should we consider it as a design-system component that will be reusable in the feature?

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/289ce951-03d4-4497-a811-6bd9cba9bdd3)

6. Would be nice to have mobile design as well for 3rd step.

**Comment by sachben91** [comment]
> @sachben91 Questions:
> 
> 1. > Namespace should be autogenerated by default to be the community name that the user entered in basic information page.
> We will re-use a util that we use for generating community ID in previous step. ID is used for the community URL field. So basically, if community name is `Warriors` then ID is `warriors`. If community name is `Super Dao` then ID is `super-dao` etc. ![image](https://private-user-images.githubusercontent.com/14819225/289077969-a186ed00-1357-471c-aa59-141b2dd3adce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNjA5MjksIm5iZiI6MTcwMjA2MDYyOSwicGF0aCI6Ii8xNDgxOTIyNS8yODkwNzc5NjktYTE4NmVkMDAtMTM1Ny00NzFjLWFhNTktMTQxYjJkZDNhZGNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDE4MzcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3M2IwMmIzNmEwZDBmZGE0YTNhNTc3NDRiYmVlMTA0MjU1ZWU1ZjE0YTU1N2FkZDU3YmM2YzNhODExMjhlOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LGzV81cUrJd413GBoesC-KWF8Hln9dYsUPe-vJRSfiI)
makes sense 
> 2. > The rules for symbols should be the same as it is right now
> 
> The only validation we have right now is max 9 characters. This means we can default to 4 first letters but then user can modify it and type any other string with any alphanumeric character, special character and white spaces. Is this ok?
This sounds good 
> 3. We have 4 different states in `Sign transactions to launch stakes` [shown here in order](https://www.figma.com/file/WLN4nRpVGQJ9AjuxiJwuDS/Create-Community?type=design&node-id=1019%3A2958&mode=dev). Does cancel button look & work the same way at every state? What happens on clicking cancel button? Maybe when the transaction is being signed we can disabled it, as it is tempting to click the button while waiting for something. Also, do we want to display any "warning modal" on clicking cancel eg when one of transaction has already signed or is being signed, similarly what we do for canceling second step
>    ![image](https://private-user-images.githubusercontent.com/14819225/289095936-c471294d-2129-4f84-ba8d-f37debdc1952.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNjA5MjksIm5iZiI6MTcwMjA2MDYyOSwicGF0aCI6Ii8xNDgxOTIyNS8yODkwOTU5MzYtYzQ3MTI5NGQtMjEyOS00Zjg0LWJhOGQtZjM3ZGViZGMxOTUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDE4MzcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5YzMxODYxZjBjMzc4MTliMzZjYTBlODZhODY4NTg5ZmM2ZmI5Y2YyZTAyMGYxM2YxMTEzM2FiMzAyMzRmNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LOP_Zks5VX7CQ1I2P-byccpYqOI4XxDlVunKetM2zdE)

- disabling cancel when transaction is being signed makes sense. If the user clicks cancel during the second step of Community Stakes they should be shown a warning modal "Are you sure you want to cancel? Community Stakes has not been enabled for your community yet". Do we need designs for this or do we have something we can reuse? 

> 4. We don't have currently this type of loader in the app. This is `CircleNotch` phosphor icon. Should it just spin? Should we consider it as a design-system component that will be reusable in the feature?
- Jared to answer
> ![image](https://private-user-images.githubusercontent.com/14819225/289098014-289ce951-03d4-4497-a811-6bd9cba9bdd3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNjA5MjksIm5iZiI6MTcwMjA2MDYyOSwicGF0aCI6Ii8xNDgxOTIyNS8yODkwOTgwMTQtMjg5Y2U5NTEtMDNkNC00NDk3LWE4MTEtNmJkOWNiYTliZGQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDE4MzcwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjZTcwYWY3ZDg2OTU1YjRhYjY3ZDE5NDUwMWM4NGNmNTJhNGM5MGM5N2ZjNWQwYmU1OTI0ZTcyYWI3MGIzN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1Z0fNl_pjr2s09XFTGPbcI5HyRF39vjAjyfCdWZ1H6Y)

> 6. Would be nice to have mobile design as well for 3rd step.

- Jared to answer
